### PR TITLE
#163801300 Reduce code complexities (smells)

### PR DESCRIPTION
**Pull request by artorious**
#### What does this PR do?

* Removes repetitions and complexities in code
#### How the app should be tested

On Postaman App: [Run in Postman](https://app.getpostman.com/run-collection/016419570361ff8cce12)

On a browswer, [visit](https://documenter.getpostman.com/view/3796196/RztoKnTh)

On a  terminal (Linux):
* Install and set up git, python3 and a virtual environment

* In a virtual environment,

  * Clone the repository `git clone https://github.com/artorious/politico-electoral-commission-api.git`

  * Run `git checkout develop`

  * Run `cd politico-electoral-commission-api`

  * Run `pip3 install -r requirements.txt` to install dependancies
  * Run `python3 run.py`
  * Test on postman

#### Any background context you want to provide?
The application is a Restful Flask API

### CODE REVIEW

**Comment by jaystaks** [comment]
Great, you've implemented...

**Comment by sportview** [comment]
your PR requests is okey

**Comment by Nduhiu17** [comment]
Nice job

**Comment by BRIGHTON-ASUMANI** [comment]
Good job you have made the necessary changes

File: app/api/v1/models/party_models.py [hunk 1]
```diff
@@ -140,3 +147,14 @@ def delete_party(pid):
             {"message": f"Party No. {pid} deleted succesfully"}
             ]
         }
+
+    def party_reg_validadion(self):
```
Comment (Mutuba) [inline app/api/v1/models/party_models.py]: @artorious , I think the function name should be party_reg_validation.

File: app/api/v1/models/office_models.py [hunk 2]
```diff
@@ -111,3 +118,16 @@ def fetch_an_office(pid):
         """ Fetch a political office by ID"""
         global POLITICAL_OFFICES
         return [office for office in POLITICAL_OFFICES if office['id'] == pid]
+
+    def office_reg_validadion(self):
```
Comment (Mutuba) [inline app/api/v1/models/office_models.py]: @artorious, kindly check the function name. Should be office_reg_validation()

File: app/api/v1/views/office_views.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from app.api.v1.models.office_models import PoliticalOffices
+from app.api.v1.views.response_vars import (
+    more_data_fields_response, few_data_fields_response,
+    id_out_of_range_response, id_cannot_be_zero_response
+)
+
+OFFICE_BP_V1 = Blueprint("v1_office", __name__, url_prefix="/api/v1")
+
+
+@OFFICE_BP_V1.route("/offices", methods=["GET"])
```
Comment (Mutuba) [inline app/api/v1/views/office_views.py:14]: You can reduce the complexity by creating a list of methods, for instance, GET, DELETE and PATCH use ids, then match function to either action using an `if`

File: app/api/v1/models/office_models.py (hunk 2 above)
Comment (artorious) [reply app/api/v1/models/office_models.py]: Hi, I've made this change

**Comment by artorious** [comment]
Hi, @Mutuba, @jaystaks, @sportview, @Nduhiu17 & @BRIGHTON-ASUMANI Thank you for the review and input. I have made changes to reflect feedback.

File: app/api/v1/models/party_models.py (hunk 1 above)
Comment (artorious) [reply app/api/v1/models/party_models.py]: Hi, I've made this change

File: app/api/v1/views/office_views.py (hunk 3 above)
Comment (artorious) [reply app/api/v1/views/office_views.py:14]: Hi, I've refactored view. I've added a helper function to perform repeated tasks

**Pull request merged**